core/src/main/java/com/novoda/merlin/receiver/ConnectivityReceiver.java

```diff
@@ -23,28 +23,33 @@ public MerlinsBeard createMerlinsBeard(Context context) {
 
         MerlinBinderRetriever merlinBinderRetriever = new MerlinBinderRetriever() {
             @Override
-            public IBinder retrieveMerlinLocalServiceBinderIfAvailable(Context context) {
-                return peekService(context, new Intent(context, MerlinService.class));
+            public MerlinService.ConnectivityChangesNotifier retrieveConnectivityChangesNotifierIfAvailable(Context context) {
+                IBinder iBinder = peekService(context, new Intent(context, MerlinService.class));
+                if (iBinder != null
+                        && iBinder instanceof MerlinService.ConnectivityChangesNotifier) {
+                    return (MerlinService.ConnectivityChangesNotifier) iBinder;
+                }
+                return null;
             }
         };
 
         ConnectivityChangeEventCreator creator = new ConnectivityChangeEventCreator();
-        connectivityChangeNotifier = new ConnectivityChangeNotifier(merlinsBeardCreator, merlinBinderRetriever, creator);
+        connectivityReceiverConnectivityChangeNotifier = new ConnectivityReceiverConnectivityChangeNotifier(merlinsBeardCreator, merlinBinderRetriever, creator);
     }
 
-    ConnectivityReceiver(ConnectivityChangeNotifier connectivityChangeNotifier) {
-        this.connectivityChangeNotifier = connectivityChangeNotifier;
+    ConnectivityReceiver(ConnectivityReceiverConnectivityChangeNotifier connectivityReceiverConnectivityChangeNotifier) {
+        this.connectivityReceiverConnectivityChangeNotifier = connectivityReceiverConnectivityChangeNotifier;
     }
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        connectivityChangeNotifier.notify(context, intent);
+        connectivityReceiverConnectivityChangeNotifier.notify(context, intent);
     }
 
     interface MerlinBinderRetriever {
 
         @Nullable
-        IBinder retrieveMerlinLocalServiceBinderIfAvailable(Context context);
```

Comment:
Not sure about the project convention but `Nullable` signature already signals nullity, not sure if `IfAvailable` suffix is necessary.

Reply:
Good point, think I will remove it 😄